atlasdb-container-test-utils/src/main/java/com/palantir/atlasdb/containers/CassandraResource.java

```diff
@@ -50,21 +50,16 @@ public CassandraResource(Supplier<KeyValueService> supplier) {
     }
 
     @Override
-    public Statement apply(Statement base, Description description) {
-        containers = new Containers(description.getTestClass()).with(containerInstance);
-        testResourceManager = new TestResourceManager(supplier);
-        return super.apply(base, description);
-    }
-
-    @Override
-    public void before() throws Throwable {
-        containers.before();
```

Comment:
The intended use of `ExternalResource` is to override the `before` and `after` methods instead of `apply`. However, here we override all of them. As far as I can see, `apply` runs first, followed by `before`. So, I preserved this order in the `beforeAll` invocation (In JUnit5, we only have `beforeAll`, and there is no equivalent to `apply`). I believe we overrode `apply` here just to use `description.getTestClass()`. We can achieve the same with `ExtensionContext` by using only `beforeAll`. Even if there are any other reasons to use `apply` and `before` separately, the tests are safe and sound. However, it would be nice to double-check my reasoning.